demisto_sdk/commands/download/downloader.py

```diff
@@ -0,0 +1,510 @@
+                if content_object:
+                    self.pack_content[content_entity].append(content_object)
+
+    def build_pack_content_object(self, content_entity: str, entity_instance_path: str) -> dict:
+        """
+        Build the pack content object the represents an entity instance.
+        For example: HelloWorld Integration in Packs/HelloWorld.
+        """
+        file_paths: list = get_files_in_dir(entity_instance_path, CONTENT_FILE_ENDINGS, recursive=False)
+        # In pack if it's integration/script all the files under it should have the main details of the yml file,
+        # else we'll use the file's details.
+        content_object: dict = dict()
+
+        if file_paths:
+            main_id, main_name = self.get_main_file_details(content_entity, entity_instance_path)
+            if main_id and main_name:
+                content_object = {main_name: list()}
+
+                for file_path in file_paths:
+                    content_object[main_name].append({
+                        'name': main_name,
+                        'id': main_id,
+                        'path': file_path,
+                        'file_ending': retrieve_file_ending(file_path)
+                    })
+
+        return content_object
+
+    @staticmethod
+    def get_main_file_details(content_entity: str, entity_instance_path: str) -> tuple:
+        """
+        Returns the details of the "main" file within an entity instance.
+        For example: In the HelloWorld integration under Packs/HelloWorld, the main file is the yml file.
+        It contains all relevant ids and names for all the files under the HelloWorld integration dir.
+        """
+        main_file_data: dict = dict()
+        main_file_path: str = str()
+
+        # Entities which contain yml files
+        if content_entity in (INTEGRATIONS_DIR, BETA_INTEGRATIONS_DIR, SCRIPTS_DIR, PLAYBOOKS_DIR, TEST_PLAYBOOKS_DIR):
+            if os.path.isdir(entity_instance_path):
+                _, main_file_path = get_yml_paths_in_dir(entity_instance_path)
+            elif os.path.isfile(entity_instance_path):
+                main_file_path = entity_instance_path
+
+            if main_file_path:
+                main_file_data = get_yaml(main_file_path)
+
+        # Entities which are json files (md files are ignored - changelog/readme)
+        else:
+            if os.path.isfile(entity_instance_path) and entity_instance_path.endswith('.json'):
+                main_file_data = get_json(entity_instance_path)
+
+        main_id = get_id_by_content_entity(main_file_data, content_entity)
+        main_name = get_name_by_content_entity(main_file_data, content_entity)
+
+        return main_id, main_name
+
+    def fetch_custom_content(self) -> None:
+        """
+        Fetches the custom content from Demisto into a temporary dir.
+        """
+        self.client = demisto_client.configure(verify_ssl=not self.insecure)
```

Comment:
It makes issues with unit testing, at the beginning it was in the init func and then to make unit tests easier I've moved it here since it's the only place we use the client.